client/src/components/HomePage/HomepageProfileModal.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useRef } from "react";
+import "./profileModal.css";
+import { userLogout } from "../../utils/authControl";
+import { useNavigate } from "react-router";
+
+export default function HomepageProfileModal({ closeModal, position }) {
+  const modalRef = useRef(null);
+  const Navigate = useNavigate();
+
+  // mock data for styling
+  const userData = {
+    name: "Dijon Miller",
+    email: "dijon.miller@example.com",
+    initials: "DM",
+    profileImage:
+      "https://img.freepik.com/free-photo/portrait-expressive-young-man-wearing-formal-suit_273609-6942.jpg?semt=ais_hybrid&w=740",
+  };
+
+  useEffect(() => {
+    // closes modal when clicked outside of modal
+    function handleClickOutside(event) {
+      if (modalRef.current && !modalRef.current.contains(event.target)) {
+        closeModal();
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [closeModal]);
+
+  // calculate position styles
+  const positionStyle = position
+    ? {
+        position: "fixed",
+        top: position.top + 20, // was blocking search bar. added 20px to move it down
```

Comment:
This is a common issue when doing styling.  It would be great to make it more "smart" by using variables

```
const searchBarStyle = {
  height: 20,
  padding: 3,
}

...

const positionStyle = {
  // padding * 2 for top and bottom
  top: searchBarStyle.height + (searchBarStyle.padding * 2)
}
```

but thats just a styling thing obviously, just an FYI